experimental/internal/oxford.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import "fmt"
+
+// Oxford is a formattable value that prints a slice joined with commas, but
+// taking care to join them with the given conjunction (such as "and") and
+// ensuring that an Oxford comma is included, but only when necessary.
+type Oxford[T any] struct {
```

Comment:
Well, one option is we can simply give `taxa.Set` a `.Join(conj string)` method, since basically everything that will ever go into it will be a `taxa.Set` (I use this in quite a few places in the legalizer).

Yes, expect a followup to contain `taxa`.